### PR TITLE
we can freeze graph before saving

### PR DESCRIPTION
**Pull request by SoluMilken**
Issue closed https://github.com/Yoctol/serving-utils/issues/25 ?

### CODE REVIEW

File: serving_utils/saver.py [hunk 1]
```diff
@@ -43,3 +39,25 @@ def save(
...
+            input_graph_def=self.session.graph_def,
+            output_node_names=output_op_names,
+        )
+
+        new_sess = tf.Session(graph=tf.Graph())
+        with new_sess.graph.as_default():
+            tf.import_graph_def(frozen_graph_def, name="")
+
+        # update session
+        self.session.close()
```
Comment (stegben) [inline serving_utils/saver.py]: 別幫別人 close()

File: serving_utils/saver.py [hunk 2]
```diff
@@ -43,3 +39,25 @@ def save(
...
         builder.save()
         return output_version_dir
+
+    def freeze_graph(self):
+        """Freeze Graph and update session"""
+
+        output_op_names = []
+        for signature_value in self.signature_def_map.values():
+            for _, tensor_info in signature_value.outputs.items():
+                output_op_names.append(tensor_info.name[:-2])
```
Comment (stegben) [inline serving_utils/saver.py]:
```suggestion
                tensor_name = tensor_info.name
                if ':' in tensor_name:
                    op_name = tensor_name.split(':')[0]
                else:
                    op_name = tensor_name
                output_op_names.append(op_name)
                
```

Comment (GBLin5566) [inline serving_utils/saver.py]: what's ```-2``` for ?

Comment (SoluMilken) [reply serving_utils/saver.py]: Tensor name = Operation name + ":0"

File: serving_utils/saver.py (hunk 1 above)
Comment (SoluMilken) [reply serving_utils/saver.py]: 那要直接 assign 新的給他嗎 @@

File: serving_utils/saver.py (hunk 2 above)
Comment (SoluMilken) [reply serving_utils/saver.py]: f94de0b

File: serving_utils/saver.py (hunk 1 above)
Comment (SoluMilken) [reply serving_utils/saver.py]: f94de0b

File: serving_utils/saver.py (hunk 2 above)
Comment (SoluMilken) [reply serving_utils/saver.py]: better one
f94de0b

File: serving_utils/saver.py [hunk 3]
```diff
@@ -11,35 +12,67 @@ def __init__(
...
             builder = tf.saved_model.builder.SavedModelBuilder(
                 export_dir=output_version_dir,
             )
             builder.add_meta_graph_and_variables(
-                sess=self.session,
+                sess=session_to_be_saved,
                 tags=[tf.saved_model.tag_constants.SERVING],
                 signature_def_map=self.signature_def_map,
             )
         builder.save()
```
Comment (stegben) [inline serving_utils/saver.py]: 也許加一個
```python
if self.freeze:
    session_to_be_saved.close()
```
以釋放資源

**Comment by stegben** [comment]
順便加一個用法在 README 吧

File: serving_utils/saver.py (hunk 3 above)
Comment (SoluMilken) [reply serving_utils/saver.py]: :ok_woman:

**Comment by SoluMilken** [comment]
先別merge

**Pull request merged**